quil-py/src/program/scheduling.rs

```diff
@@ -0,0 +1,86 @@
+use quil_rs::program::scheduling::{
+    ComputedScheduleItem, FixedSchedule, ScheduledBasicBlock, ScheduledBasicBlockOwned, Seconds,
+    TimeSpan,
+};
+use rigetti_pyo3::{impl_repr, py_wrap_type, pyo3::prelude::*};
+
+use super::PyProgram;
+
+py_wrap_type! {
+    PyScheduledBasicBlock(ScheduledBasicBlockOwned) as "ScheduledBasicBlock"
+}
+
+impl_repr!(PyScheduledBasicBlock);
+
+impl PyScheduledBasicBlock {
+    pub fn as_fixed_schedule(&self, program: &PyProgram) -> PyFixedSchedule {
+        ScheduledBasicBlock::from(&self.0)
+            .as_fixed_schedule(&program.0)
+            .expect("todo handle error")
+            .into()
+    }
+}
+
+py_wrap_type! {
+    PyFixedSchedule(FixedSchedule) as "FixedSchedule"
+}
+
+impl_repr!(PyFixedSchedule);
+
+#[pymethods]
+impl PyFixedSchedule {
+    pub fn items(&self) -> Vec<PyFixedScheduleItem> {
+        self.0
+            .items()
+            .iter()
+            .map(PyFixedScheduleItem::from)
+            .collect()
+    }
+
+    pub fn duration(&self) -> f64 {
+        self.0.duration().0
+    }
+}
+
+py_wrap_type! {
+    PyFixedScheduleItem(ComputedScheduleItem<Seconds>) as "FixedScheduleItem"
+}
+
+impl_repr!(PyFixedScheduleItem);
+
+#[pymethods]
+impl PyFixedScheduleItem {
+    #[getter]
+    pub fn time_span(&self) -> PyFixedTimeSpan {
+        (&self.0.time_span).into()
+    }
+
+    #[getter]
+    pub fn instruction_index(&self) -> usize {
+        self.0.instruction_index
+    }
+}
+
+py_wrap_type! {
+    PyFixedTimeSpan(TimeSpan<Seconds>) as "FixedTimeSpan"
```

Comment:
I think it would be nice to include the unit in the type somewhere, e.g. `PyFixedTimeSpanSeconds`, even though I know it's mentioned in the docstrings.

Reply:
True, I should do the same on the Rust side.